components/valve/src/main/java/org/wso2/carbon/extension/identity/x509Certificate/valve/config/X509ServerConfiguration.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.extension.identity.x509Certificate.valve.config;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import javax.xml.namespace.QName;
+
+public class X509ServerConfiguration {
+
+    private static Log log = LogFactory.getLog(X509ServerConfiguration.class);
+    private static final X509ServerConfiguration INSTANCE = new X509ServerConfiguration();
+    private static final String CONFIG_ELEM_X509 = "X509";
+    private static final String CONFIG_ELEM_X509_REQUEST_HEADER = "X509RequestHeaderName";
+
+    private String x509requestHeader = "X-SSL-CERT";
+
+    private X509ServerConfiguration() {
+
+        buildX509ServerConfiguration();
+    }
+
+    public static X509ServerConfiguration getInstance() {
+
+        CarbonUtils.checkSecurity();
+        return INSTANCE;
+    }
+
+    /**
+     * @return name of the X509 request header
+     */
+    public String getX509requestHeader() {
+
+        return x509requestHeader;
+    }
+
+    private void buildX509ServerConfiguration() {
+
+        IdentityConfigParser configParser = IdentityConfigParser.getInstance();
```

Comment:
you can directly use IdentityUtil.getProperty("X509.X509RequestHeaderName") instead of using the parser

Reply:
Since this is not an osgi bundle, lit is better to use IdentityConfigParser. auth-rest-valve also uses IdentityConfigParser